src/xtopcom/xtopcl/src/api_method.cpp

```diff
@@ -685,8 +690,40 @@ void ApiMethod::query_miner_info(std::string & account, std::ostringstream & out
     string null_out = g_userinfo.account + " account has not registered miner.";
     tackle_null_query(out_str, null_out);
 }
-
-void ApiMethod::register_node(const double & mortgage_d,
+int ApiMethod::parse_top_double(const std::string &amount, const uint32_t unit, uint64_t &out)
```

Comment:
may need more checks again invalid `amount` input. PM should be involved in.

Reply:
![image](https://user-images.githubusercontent.com/2065762/118109279-898c8c80-b413-11eb-90d5-b61d09f7a655.png)